vasyl_varyshchuk/Week_3/google_search_request.py

```diff
@@ -0,0 +1,76 @@
+"""
+This module sends a request to google.com returning the response headers
+and the specified number of searching results
+"""
+import requests
+from pip import main
+from pip.req import parse_requirements
+
+
+def install_required_modules():
+    """This function performs installation of required modules from requirements.txt file"""
+    required_modules = parse_requirements('requirements.txt', session=False)
+    requirements = [str(i.req) for i in required_modules]
+    for requirement in requirements:
+        main(['install', requirement])
+
+
+try:
+    from bs import BeautifulSoup
+except ImportError:
+    user_answer = None
```

Comment:
It's odd block of code.

It's enough to create `requirements.txt` file and handle `ImportError`.
In your handler you should write a message like `This script requires `bs` module. Please install it using `requirements.txt` or something similar. That's it.

Reply:
I realized. I changed handler for ImportError to avoid redundancy in the code.